src/client/QXmppMixManager.cpp

```diff
@@ -0,0 +1,1382 @@
+///
+
+///
+/// Returns whether the own server supports archiving messages via
+/// \xep{0313, Message Archive Management} of MIX channels the user participates in.
+///
+/// \return whether MIX messages are archived
+///
+bool QXmppMixManager::archivingSupportedByServer() const
+{
+    return d->archivingSupportedByServer;
+}
+
+///
+/// \fn QXmppMixManager::archivingSupportedByServerChanged()
+///
+/// Emitted when the server enabled or disabled supporting archiving for MIX.
+///
+
+///
+/// Returns the services providing MIX on the own server.
+///
+/// Such services provide MIX channels and their nodes.
+/// It interacts directly with clients or with their servers.
+///
+/// \return the provided MIX services
+///
+QList<QXmppMixManager::Service> QXmppMixManager::services() const
+{
+    return d->services;
+}
+
+///
+/// \fn QXmppMixManager::servicesChanged()
+///
+/// Emitted when the services providing MIX on the own server changed.
+///
+
+///
+/// Creates a MIX channel.
+///
+/// If no channel ID is passed, the channel is created with an ID provided by the MIX service.
+/// Furthermore, the channel cannot be discovered by anyone.
+/// A channel with the mentioned properties is called an "ad-hoc channel".
+///
+/// The channel ID is the local part of the channel JID.
+/// The MIX service JID is the domain part of the channel JID.
+/// Example: "channel" is the channel ID and "mix.example.org" the service JID of the channel JID
+/// "channel@mix.example.org".
+///
+/// \param serviceJid JID of the service
+/// \param channelId ID of the channel (default: provided by the server)
+///
+/// \return the result of the action
+///
+QXmppTask<QXmppMixManager::CreationResult> QXmppMixManager::createChannel(const QString &serviceJid, const QString &channelId)
+{
+    QXmppMixIq iq;
+    iq.setType(QXmppIq::Set);
+    iq.setTo(serviceJid);
+    iq.setActionType(QXmppMixIq::Create);
+    iq.setChannelId(channelId);
+
+    return chainIq(client()->sendIq(std::move(iq)), this, [](QXmppMixIq &&iq) -> CreationResult {
+        return iq.channelJid().isEmpty() ? iq.channelId() % u'@' % iq.from() : iq.channelJid();
+    });
+}
+
+///
+/// Requests the JIDs of all discoverable MIX channels of a MIX service.
+///
+/// \param serviceJid JID of the service that provides the channels
+///
+/// \return the result of the action
+///
+QXmppTask<QXmppMixManager::ChannelJidResult> QXmppMixManager::requestChannelJids(const QString &serviceJid)
+{
+    return chain<ChannelJidResult>(d->discoveryManager->requestDiscoItems(serviceJid), this, [](QXmppDiscoveryManager::ItemsResult &&result) -> ChannelJidResult {
+        if (const auto items = std::get_if<QList<QXmppDiscoveryIq::Item>>(&result)) {
+            QVector<ChannelJid> jids;
+
+            std::for_each(items->cbegin(), items->cend(), [&jids](const QXmppDiscoveryIq::Item &item) {
+                jids.append(item.jid());
+            });
+
+            return std::move(jids);
```

Comment:
```suggestion
            return transform<QVector<ChannelJid>>(items, [](const auto &item) { return item.jid(); });
```
There's a transform helper function i used in some places, I moved it into an Algorithms.h header now (as in kaidan).
See https://github.com/qxmpp-project/qxmpp/commit/9d728b64c553a19b7bbe26c520041850dc2c8049